frigate/object_detection.py

```diff
@@ -23,46 +25,47 @@ def detect(self, tensor_input, threshold=0.4):
         pass
 
 
+def tensor_transform(desired_shape):
+    # Currently this function only supports BHWC permutations
+    if desired_shape == ["B", "H", "W", "C"]:
+        return None
+    else:
+        transform = [0] * 4
+        transform[desired_shape.index("H")] = 1
+        transform[desired_shape.index("W")] = 2
+        transform[desired_shape.index("C")] = 3
+        return tuple(transform)
+
+
 class LocalObjectDetector(ObjectDetector):
-    def __init__(self, tf_device=None, model_path=None, num_threads=3, labels=None):
+    def __init__(
+        self,
+        det_type=DetectorTypeEnum.cpu,
+        det_device=None,
+        model_config=None,
+        num_threads=3,
+        labels=None,
+    ):
         self.fps = EventsPerSecond()
         if labels is None:
             self.labels = {}
         else:
             self.labels = load_labels(labels)
 
-        device_config = {"device": "usb"}
-        if not tf_device is None:
-            device_config = {"device": tf_device}
-
-        edge_tpu_delegate = None
-
-        if tf_device != "cpu":
-            try:
-                logger.info(f"Attempting to load TPU as {device_config['device']}")
-                edge_tpu_delegate = load_delegate("libedgetpu.so.1.0", device_config)
-                logger.info("TPU found")
-                self.interpreter = tflite.Interpreter(
-                    model_path=model_path or "/edgetpu_model.tflite",
-                    experimental_delegates=[edge_tpu_delegate],
-                )
-            except ValueError:
-                logger.error(
-                    "No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors."
-                )
-                raise
+        if model_config:
```

Comment:
Is this supposed to be `self.model_config.input_tensor` ?

self.model_config will always be non-null since it is built by default even if a user doesn't define it in their config

Reply:
I see there's a couple places where LocalObjectDetector is instantiated without passing in a model.  This check will save us from an error on dereferencing `model_config` if it is `None` and the `tensor_transform()` function will do another check on the argument.

Reply:
Yeah I just missed those other locations as I was just looking in app